sunpy/map/mapbase.py

```diff
@@ -243,7 +249,66 @@ def __repr__(self):
                                refpix=u.Quantity(self.reference_pixel),
                                refcoord=u.Quantity((self.reference_coordinate.data.lon,
                                                     self.reference_coordinate.data.lat)),
-                               tmf=TIME_FORMAT) + self.data.__repr__()
+                               tmf=TIME_FORMAT)
+
+    def __repr__(self):
+        return object.__repr__(self) + "\n" + self._text_summary() + "\n" + self.data.__repr__()
+
+    def _repr_html_(self):
+        """
+        Produce an HTML summary with plots for use in Jupyter notebooks.
+        """
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 4.8), constrained_layout=True)
+        ax = fig.subplots()
+        cmap = cm.get_cmap('gray')
+        cmap.set_bad(color='red')
+        ax.imshow(self.data, cmap=cmap, origin='lower',
+                  norm=ImageNormalize(stretch=HistEqStretch(self.data[~np.isnan(self.data)])))
+        ax.set_xlabel('X pixel')
+        ax.set_ylabel('Y pixel')
+        ax.set_title('Histogram-equalized image' + (' (NaN values in red)' if np.isnan(self.data).sum() else ''))
+        fig.savefig(buf, format='png')
+        image_src = b64encode(buf.getvalue()).decode('utf-8')
+
+        buf = BytesIO()
+        fig = Figure(figsize=(4.8, 2.4), constrained_layout=True)
+        ax = fig.subplots()
+        ax.hist(self.data.ravel(), bins=100, histtype='stepfilled')
+        ax.semilogy()
+        ax.xaxis.get_major_formatter().set_powerlimits((-3, 4))
+        ax.set_xlabel('Pixel value')
+        ax.set_ylabel('# of pixels')
+        fig.savefig(buf, format='png')
```

Comment:
I think this is where you can give this arg:
```suggestion
        fig.savefig(buf, format='png', transparent=True)
```

Reply:
`transparent=True` seems to work as desired for the images, but not for this histogram...  I don't know what's going on.

Reply:
It turns out that `transparent=True` was not only working, but working too well and overriding other colors.  Anyway, fixed now.